### PR TITLE
coap-gw: set not found for empty body from batch observation

### LINKED ISSUE #766: Recreated resource is not republished
A resource was created with specific attributes and values. Then it was deleted and re-created again with exact same properties as before.
It was observed that resource was not republished.

If the resource is re-created but with a different value in some attribute, it is republished.

### PR DESCRIPTION
**Pull request by jkralik**
ra: set not found to unpublished resources

Fixes #766

### CODE REVIEW

**Review by ondrejtomcik — APPROVED** [review]

File: grpc-gateway/service/createAndDeleteResource_test.go [hunk 1]
```diff
@@ -281,15 +270,14 @@ func TestCreateAndDeleteResource(t *testing.T) {
 	subClient, subID := subscribeToAllEvents(t, ctx, c, correlationID)
 	const switchID = "1"
 
-	isDiscoveryResourceBatchObservable := test.IsDiscoveryResourceBatchObservable(ctx, t, deviceID)
-
 	for i := 0; i < 5; i++ {
 		fmt.Printf("iteration %v\n", i)
+		time.Sleep(time.Millisecond * 500)
```
Comment (Danielius1922) [inline grpc-gateway/service/createAndDeleteResource_test.go:275]: Sleep after print seems weird, is the placement correct? Usually we delay after a create/delete call.

Comment (jkralik) [reply grpc-gateway/service/createAndDeleteResource_test.go:275]: As it is async, unpublish events can still be processed during resource creation. (some time the iotivity-lite test fails so I needed add there some time to get idle state)

File: resource-aggregate/events/resourceStateSnapshotTaken.go [hunk 2]
```diff
@@ -280,7 +280,6 @@ func (e *ResourceStateSnapshotTaken) HandleEventResourceChanged(ctx context.Cont
 
 func (e *ResourceStateSnapshotTaken) HandleEventResourceDeleted(ctx context.Context, deleted *ResourceDeleted) error {
 	if deleted.GetStatus() == commands.Status_OK || deleted.GetStatus() == commands.Status_ACCEPTED {
-		e.LatestResourceChange = nil
```
Comment (Danielius1922) [inline resource-aggregate/events/resourceStateSnapshotTaken.go:283]: I recall adding this line a few months ago, won't it be a problem if we keep the resource change after the resource is deleted?

Comment (jkralik) [reply resource-aggregate/events/resourceStateSnapshotTaken.go:283]: When resource has been deleted we expect two same events that set `LatestResourceChange`:

1. unpublish event generates `ResourceContentChanged` event with status `NotFound`
2. cancellation of observation  generates `ResourceContentChanged` event with status `NotFound`

It depends which comes first do this is why it need to be same for deduplication.

**Review by Danielius1922 — APPROVED** [review]

**Pull request merged**